gadget/install/install.go

```diff
@@ -160,14 +173,32 @@ func Run(gadgetRoot, device string, options Options, observer SystemInstallObser
 		}
 	}
 
+	dataKeySet := keysForRoles[gadget.SystemData]
+	if dataKeySet == nil {
+		return fmt.Errorf("internal error: system data encryption key set is unset")
+	}
+
 	// Write the recovery key
 	recoveryKeyFile := filepath.Join(boot.InstallHostFDEDataDir, "recovery.key")
-	if err := rkey.Save(recoveryKeyFile); err != nil {
+	if err := dataKeySet.rkey.Save(recoveryKeyFile); err != nil {
 		return fmt.Errorf("cannot store recovery key: %v", err)
 	}
 
 	if observer != nil {
-		observer.ChosenEncryptionKey(key)
+		observer.ChosenEncryptionKey(dataKeySet.key)
+	}
+
+	saveKeySet := keysForRoles[gadget.SystemSave]
+	if saveKeySet != nil {
+		saveKey := filepath.Join(boot.InstallHostFDEDataDir, "ubuntu-save.key")
+		reinstallSaveKey := filepath.Join(boot.InstallHostFDEDataDir, "reinstall.key")
```

Comment:
Should this be ubuntu-save-recovery.key" or something?

Reply:
@pedronis suggested `reinstall.key` in https://github.com/snapcore/snapd/pull/9443#discussion_r498275845